### PR TITLE
Add XYZ reference to Pulmap Solr document

### LINKED ISSUE #4885: Add XYZ reference to Pulmap Solr Document for RasterResources

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #4885 
Closed https://github.com/pulibrary/figgy/issues/4897

### CODE REVIEW

File: app/services/geo_discovery/document_builder/wxs.rb [hunk 1]
```diff
@@ -43,8 +43,17 @@ def wcs_path
...
       # @return [String] wmts server url
       def wmts_path
-        return unless resource_decorator.object.decorate.try(:raster_set?)
-        "#{tileserver_path}/mosaicjson/WMTSCapabilities.xml?id=#{resource_decorator.id}"
+        return unless visibility == "open" && (raster_set? || raster_file_set?)
+        id = resource_decorator.id.to_s.delete("-")
+        "#{tileserver_path}/mosaicjson/WMTSCapabilities.xml?id=#{id}"
+      end
+
+      # Returns the wmts server url.
```
Comment (hackartisan) [inline app/services/geo_discovery/document_builder/wxs.rb:51]: need to update these comments

File: spec/services/geo_discovery/document_builder_spec.rb [hunk 2]
```diff
@@ -82,7 +82,7 @@
       expect(refs["http://schema.org/downloadUrl"]).to match(/downloads/)
       expect(refs["http://www.opengis.net/def/serviceType/ogc/wms"]).to match(/geoserver\/public-figgy\/wms/)
       expect(refs["http://www.opengis.net/def/serviceType/ogc/wfs"]).to match(/geoserver\/public-figgy\/wfs/)
-      # wmts only exists for a raster set
+      # wmts only exists for a rasters and raster sets
```
Comment (hackartisan) [inline spec/services/geo_discovery/document_builder_spec.rb:85]: There's an extra 'a' here.

**Review by hackartisan — CHANGES_REQUESTED** [review]
This looks good, with a couple of small things in line.

questions:
- wmts is for rasters that aren't part of a mosaic, also?
- why are we removing the other tile service? It would be nice to have an issue for this to point to.

**Comment by eliotjordan** [comment]
@hackartisan Yes, individual rasters can (and should) be served out using TiTiler WMTS and XYZ endpoints. I created an issue for the tile_layer bit:
https://github.com/pulibrary/figgy/issues/4897

Rediscovered it as I was working this PR and it made sense to get rid of at the same time.

**Review by hackartisan — APPROVED** [review]

**Pull request merged**